lib/wanda/results.ex

```diff
@@ -5,25 +5,21 @@ defmodule Wanda.Results do
 
   alias Wanda.Repo
 
-  alias Wanda.Execution.Result
+  alias Wanda.Execution.{Result, Target}
   alias Wanda.Results.ExecutionResult
 
   import Ecto.Query
 
   @doc """
-  Create a new result.
+  Create a new execution as soon as it starts.
   """
-  @spec create_execution_result!(Result.t()) :: ExecutionResult.t()
-  def create_execution_result!(
-        %Result{
-          execution_id: execution_id,
-          group_id: group_id
-        } = result
-      ) do
+  @spec create_execution_result!(String.t(), String.t(), [Target.t()]) :: ExecutionResult.t()
+  def create_execution_result!(execution_id, group_id, targets) do
     Repo.insert!(%ExecutionResult{
       execution_id: execution_id,
       group_id: group_id,
-      payload: result
+      status: :running,
+      targets: Enum.map(targets, &Map.from_struct(&1))
```

Comment:
I believe you don't need to map those if they have `Jason.encoder` protocol derive

Reply:
Adding `@derive Jason.Encoder` to `Wanda.Execution.Target` does not seem to be enough.

If I do that and then I pass `targets` here directly
```elixir
    Repo.insert!(%ExecutionResult{
      execution_id: execution_id,
      group_id: group_id,
      status: :running,
      targets: targets
    })
```
I get this error 
```
** (UndefinedFunctionError) function Wanda.Execution.Target.__changeset__/0 is undefined or private
```
I believe because `Wanda.Execution.Target` is not an ecto struct.

Alternatively I can use `ExecutionResult.changeset` and pass `targets` to it
```elixir
    %ExecutionResult{}
    |> ExecutionResult.changeset(%{
      execution_id: execution_id,
      group_id: group_id,
      status: :running,
      targets: targets
    })
    |> Repo.insert!()
```
yet it is not enough anyway because now ecto's changeset expects a map when casting `:targets` inside an `ExecutionResult` (received error `Ecto.CastError (expected params to be a :map, got: %Wanda.Execution.Target`)

At the end, to make it work I still needed a `Map.from_struct`, but in another place

```elixir
defmodule Wanda.Results.ExecutionResult do
  ...
  defp target_changeset(target, params) do
    cast(target, Map.from_struct(params), [:agent_id, :checks])
  end
  ...
end
```

So, for what I could experiment we either have `targets: Enum.map(targets, &Map.from_struct(&1))` in the context or `cast(target, Map.from_struct(params), [:agent_id, :checks])` in the changeset.

I don't see much value in creating another specific schema atm, let alone make `Wanda.Execution.Target` an ecto schema itself.